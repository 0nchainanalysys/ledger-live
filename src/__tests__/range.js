// @flow
import {
  inferDynamicRange,
  projectRangeIndex,
  reverseRangeIndex
} from "../range";

test("inferDynamicRange", () => {
  expect(inferDynamicRange(0.4)).toMatchSnapshot();
  expect(inferDynamicRange(1)).toMatchSnapshot();
  expect(inferDynamicRange(1.01)).toMatchSnapshot();
  expect(inferDynamicRange(10)).toMatchSnapshot();
  expect(inferDynamicRange(100)).toMatchSnapshot();
  expect(inferDynamicRange(99)).toMatchSnapshot();
});

test("projectRangeIndex", () => {
  expect(projectRangeIndex(inferDynamicRange(0.4), 2)).toMatchSnapshot();
});

test("reverseRangeIndex", () => {
  const range = inferDynamicRange(0.4);
  for (let i = 0; i < range.steps; i++) {
    const n = projectRangeIndex(range, i);
    expect(reverseRangeIndex(range, n)).toBe(i);
  }
});
