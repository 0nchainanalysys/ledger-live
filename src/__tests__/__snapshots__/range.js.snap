// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`inferDynamicRange 1`] = `
Object {
  "initial": 0.4,
  "max": 0.8,
  "min": 0.1,
  "step": 0.05,
  "steps": 15,
}
`;

exports[`inferDynamicRange 2`] = `
Object {
  "initial": 1,
  "max": 2,
  "min": 0.2,
  "step": 0.1,
  "steps": 19,
}
`;

exports[`inferDynamicRange 3`] = `
Object {
  "initial": 1,
  "max": 2.1,
  "min": 0.30000000000000004,
  "step": 0.1,
  "steps": 19,
}
`;

exports[`inferDynamicRange 4`] = `
Object {
  "initial": 10,
  "max": 20,
  "min": 3,
  "step": 1,
  "steps": 18,
}
`;

exports[`inferDynamicRange 5`] = `
Object {
  "initial": 100,
  "max": 200,
  "min": 30,
  "step": 10,
  "steps": 18,
}
`;

exports[`inferDynamicRange 6`] = `
Object {
  "initial": 100,
  "max": 200,
  "min": 20,
  "step": 10,
  "steps": 19,
}
`;

exports[`projectRangeIndex 1`] = `0.2`;
